### PR TITLE
feat(hosting): common offers selector

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/web-nwo`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat #MANAGER-11054
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
1. Create a common component to manage offers

### CODE REVIEW

File: packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    >
+        <!--Offer picker header-->
+        <oui-select-picker-label class="w-100 d-flex justify-content-between">
+            <!--Offer name-->
+            <div class="m-0">
+                <span
+                    class="font-weight-bold"
+                    data-ng-bind="('web_components_hosting_domain_offers_offer_' + offer.category | translate) | uppercase"
+                ></span>
+            </div>
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html:22]: **nitpick:** `div` + `span` can be combined into `p`.
```suggestion
            <p class="m-0"
                    class="font-weight-bold"
                    data-ng-bind="('web_components_hosting_domain_offers_offer_' + offer.category | translate) | uppercase"
                >
            </p>
```

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+            <!--Offer label-->
+            <div class="m-0">
+                <span class="oui-badge oui-badge_new">New</span>
+            </div>
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html:27]: Same as https://github.com/ovh/manager/pull/9182/files#r1166506167

File: packages/manager/apps/web/client/app/components/hosting-domain-offers/translations/Messages_fr_FR.json
Comment (anooparveti) [inline packages/manager/apps/web/client/app/components/hosting-domain-offers/translations/Messages_fr_FR.json:1]: Can you replace `value1` with meaningful variable names for dynamic values?

File: packages/manager/apps/web/client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js [hunk 3]
```diff
@@ -55,7 +57,10 @@ angular.module('App').controller(
         downgradeAgree: false,
       };
 
-      this.WucUser.getUser().then((user) => {
+      const user = this.coreConfig.getUser();
+      this.Hosting.getCatalog(user.ovhSubsidiary).then((catalog) => {
+        this.user = user;
+        this.catalog = catalog;
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js:63]: **nitpick:** Temporary local variable can be avoided.
```suggestion
      this.user = this.coreConfig.getUser();
      this.Hosting.getCatalog(this.user.ovhSubsidiary).then((catalog) => {
        this.catalog = catalog;
```

File: packages/manager/apps/web/client/app/components/hosting-domain-offers/translations/Messages_fr_FR.json
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/components/hosting-domain-offers/translations/Messages_fr_FR.json:1]: no, it's must be a generic key, used for different kind of cases.

File: packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html (hunk 1 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html:22]: changed

File: packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html (hunk 2 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/components/hosting-domain-offers/template.html:27]: changed

File: packages/manager/apps/web/client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js (hunk 3 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js:63]: it avoid to accessing lot times to user object.
It used 3 times after declaration.

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**